endor/src/services/invites.service.js

```diff
@@ -0,0 +1,187 @@
+import { InviteStatus } from '../db/sequelize';
+
+import InviteNotFoundException from '../error/InviteNotFoundException';
+import InvalidRequestException from '../error/InvalidRequestException';
+import RequestParamError from '../error/RequestParamError';
+
+function isNonNegativeInteger(value) {
+  return Number.isInteger(value) && value >= 0;
+}
+
+export default class InviteService {
+  /**
+   * Invite Service constructor
+   * @param inviteRepository the invite repository to inject (most likely something from sequalize)
+   * @param log the logger to inject
+   */
+  constructor(inviteRepository, log) {
+    this.inviteRepository = inviteRepository;
+    this.log = log;
+  }
+
+
+  /**
+   * Returns true if the given status is a match to one of the possible enumerations
+   * @param status the string to check
+   * @returns {boolean} whether or not the string is a valid InviteStatus
+   */
+  static isValidInviteStatus(status) {
+    const inviteStatusValues = Object.keys(InviteStatus).map(key => InviteStatus[key]);
+    return inviteStatusValues.filter(value => value === status).length > 0;
+  }
+
+
+  /**
+   * Validates a invite
+   * @param invite the invite to validate
+   * @param asNewInvite boolean indicates whether it should be validated
+   *   as a new invite or as an update
+   * @returns {Array} list of errors
+   */
+  static async validateInvite(invite, asNewInvite) {
+    const errors = [];
+    if (asNewInvite) {
+      if (!('userInvitedId' in invite)) {
+        errors.push(new RequestParamError('userInvitedId', 'User is required.'));
+      }
+
+      if (!('projectInvitedToId' in invite)) {
+        errors.push(new RequestParamError('projectInvitedToId', 'Project is required.'));
+      }
+    } else if (!('status' in invite)) {
+      errors.push(new RequestParamError('status', 'Status is required for invite updates.'));
+    }
+
+    if (invite.daysFromCreationUntilExpiration
+      && !isNonNegativeInteger(invite.daysFromCreationUntilExpiration)) {
+      errors.push(new RequestParamError('daysFromCreationUntilExpiration', 'Must be a non-negative integer.'));
+    }
+
+    if (invite.status && !InviteService.isValidInviteStatus(invite.status)) {
+      const inviteStatusValues = Object.keys(InviteStatus).map(key => InviteStatus[key]);
+      const enumValues = inviteStatusValues.join(', ');
+      errors.push(new RequestParamError('status', `Must be one of the following: ${enumValues}`));
+    }
+
+    return errors;
+  }
```

Comment:
We can probably extract all of this to a Validator File. Then we can use the validator as middleware. That way, in a perfect world, the call to the service wouldn't even need to be made.